HAMSTR-519 Direct Pay UI

HQS-Gun opened the pull request:
**Motivation**

Direct pay is working essentially (or the essential elements of it, are working) but some UI elements are just temporarily being used right now. 


**Changes :** 

1. Fixed the "Pay Direct" button to properly reflect EVM payments with Ethereum icon instead of Bitcoin icon
2. Created a chain selection interstitial modal that appears after clicking "Pay with Crypto wallet"
3. Implemented network switching functionality in the interstitial modal


**Test:**

1. Buy a product by following the normal purchase process
2. When clicking on "Pay with Crypto wallet", verify the chain selection modal appears
3. When selecting a chain, verify the network switches properly and navigates to the payment processor page

jrkosinski reviewed (changes requested):
1. The interstitial should show only for direct payment, not pay with wallet. When paying with wallet, the user will use their currently already-connected chain. When paying direct, they get to choose a different chain.

2. The interstitial doesn't need to change their currently selected chain. That can stay the same, but just allow them to choose what chain they will send payment to.

3. If it's difficult to implement it without switching chains, then it's ok to keep it as is. But the switching of connected chain isn't necessary.

HQS-Gun commented:
**Changes:**

1. Removed automatic chain switching in wallet when selecting payment network
2. Simplified chain selection to just record user's choice without changing wallet connection
